### PR TITLE
Decouple search FAB from SearchBarScreen

### PR DESCRIPTION
**Pull request by FooIbar**

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryListScreen.kt [hunk 1]
```diff
@@ -576,6 +580,26 @@ fun GalleryListScreen(lub: ListUrlBuilder, navigator: DestinationsNavigator) = c
...
+                targetValue = if (showSearchLayout && !fabHidden) 1f else 0f,
+                animationSpec = tween(
+                    FAB_ANIMATE_TIME,
+                    if (showSearchLayout && !fabHidden) FAB_ANIMATE_TIME else 0,
+                ),
+                label = "hiddenState",
+            )
+            FloatingActionButton(
+                onClick = { onApplySearch("") },
+                modifier = Modifier.graphicsLayer {
```
Comment (revonateB0T) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryListScreen.kt:594]: We need `snackBarPadding` here

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryListScreen.kt:594]: Oops, accidentally deleted that.

**Review by revonateB0T — APPROVED** [review]

File: app/src/main/kotlin/com/hippo/ehviewer/ui/screen/SearchBarScreen.kt [hunk 2]
```diff
@@ -106,15 +97,15 @@ fun SearchBarScreen(
...
-    onApplySearch: suspend (String) -> Unit,
+    onApplySearch: (String) -> Unit,
     onSearchExpanded: () -> Unit,
     onSearchHidden: () -> Unit,
     refreshState: PullToRefreshState? = null,
     suggestionProvider: SuggestionProvider? = null,
     searchBarOffsetY: () -> Int,
     trailingIcon: @Composable () -> Unit,
     filter: @Composable (() -> Unit)? = null,
+    floatingActionButton: @Composable () -> Unit = {},
```
Comment (revonateB0T) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/screen/SearchBarScreen.kt:108]: nit: Make this default to null

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/screen/SearchBarScreen.kt:108]: `Scaffold` requires it to be non-null.

**Pull request merged**